tests/test_reification.py

```diff
@@ -56,123 +60,140 @@ def test_non_epistemic_rules(self):
         )
-
+    
     def test_epistemic_atom_with_default_negation(self):
         self.assert_equal_program(
             parse_program(":- &k{ not a}."),
-            ":- k_not_u(a). not_u(a) :- not u(a). {k_not_u(a)} :- not_u(a).",
+            ":- k(not1(u(a))). not1(u(a)) :- not u(a). {k(not1(u(a)))} :- not1(u(a)).",
         )
+        
         self.assert_equal_program(
             parse_program("b :- &k{ not a}."),
-            "u(b) :- k_not_u(a). not_u(a) :- not u(a). {k_not_u(a)} :- not_u(a).",
+            "u(b) :- k(not1(u(a))). not1(u(a)) :- not u(a). {k(not1(u(a)))} :- not1(u(a)).",
         )
+        
         self.assert_equal_program(
             parse_program(":- &k{ not not a}."),
-            ":- k_not2_u(a). not2_u(a) :- not not u(a). {k_not2_u(a)} :- not2_u(a).",
+            ":- k(not2(u(a))). not2(u(a)) :- not not u(a). {k(not2(u(a)))} :- not2(u(a)).",
         )
-
+  
     def test_epistemic_atom_with_both_negations(self):
-        pass
-        # self.assert_equal_program(parse_program(":- &k{ not -a}."), ":- k_not_sn_u(a). not_sn_u(a) :- not sn_u(a). {k_not_sn_u(a)} :- not_sn_u(a). sn_u(a) :- -u(a).")
-        # self.assert_equal_program(parse_program(":- &k{ not not -a}."), ":- k_not2_sn_u_a. not2_sn_u_a :- not not sn_u_a.  {k_not2_sn_u_a} :- not2_sn_u_a. sn_u_a :- -u_a.")
-
-    # Changed above
-    ######################################################
-    # Not changed below
+        self.assert_equal_program(
+            parse_program(":- &k{ not -a}."), 
+            ":- k(not1(u(-a))). not1(u(-a)) :- not u(-a). {k(not1(u(-a)))} :- not1(u(-a)). u(-a) :- u(-a)."
+            )
+        
+        self.assert_equal_program(
+            parse_program(":- &k{ not not -a}."),
+            ":- k(not2(u(-a))). not2(u(-a)) :- not not u(-a).  {k(not2(u(-a)))} :- not2(u(-a)). u(-a) :- u(-a)."
+            )
 
     def test_epistemic_with_variables(self):
         self.assert_equal_program(
-            parse_program(":- &k{a(V0)}."), ":- k_u(a(V0)). {k_u(a(V0))} :- u(a(V0))."
+            parse_program(":- &k{a(V0)}."), ":- k(u(a(V0))). {k(u(a(V0)))} :- u(a(V0))."
         )
+        self.assert_equal_program(
+            parse_program(":- &k{- -a(V0)}."), 
+            ":- k(u(a(V0))). {k(u(a(V0)))} :- u(a(V0)).")
+        
+        self.assert_equal_program(
+            parse_program(":- &k{ not a(V0)}."), 
+            ":- k(not1(u(a(V0)))). not1(u(a(V0))) :- not u(a(V0)). {k(not1(u(a(V0))))} :- not1(u(a(V0))).")
+        
+        self.assert_equal_program(
+            parse_program(":- &k{ not not a(V0)}."), 
+            ":- k(not2(u(a(V0)))). not2(u(a(V0))) :- not not u(a(V0)). {k(not2(u(a(V0))))} :- not2(u(a(V0))).")
+    
         self.assert_equal_program(
             parse_program(":- &k{-a(V0)}."),
-            ":- k_u(sn_a(V0)). sn_a(V0) :- -a(V0). {k_u(sn_a(V0))} :- u(sn_a(V0)).",
+            ":- k(u(-a(V0))). u(-a(V0)) :- u(-a(V0)). {k(u(-a(V0)))} :- u(-a(V0)).",
```

Comment:
In this case, do I need to introduce the u in the rule sn_a(V0) :- -a(V0) and change it to u(-a(V0)) :- u(-a(V0))?
Or should I change it to u(-a(V0)):- -a(V0)? By only changing the head of the rule? I believe it should be like this since the program is &k{-a(V0)}., but I want to double check because the implementation will change.

Reply:
Actually, this is a nice surprise.
It should be ```u(-a(V0)) :- u(-a(V0))```, but this means we can remove this rule.

Could you change the code so this rule is just not produced?